File: 백준/Silver/2606. 바이러스/바이러스.cc
#include <bits/stdc++.h>
using namespace std;

int n, m, from, to, visited[104];
vector<int> adj[104];

int dfs(int here) {
  visited[here] = 1;
  int ret = 0;

  for (int there : adj[here]) {
    if (visited[there]) continue;

    ret += 1 + dfs(there);
  }

  return ret;
}

int main() {
  cin >> n >> m;
  for (int i = 0; i < m; i++) {
    cin >> from >> to;
    adj[from].push_back(to);
    adj[to].push_back(from);
  }

  cout << dfs(1);

  return 0;
}

File: 백준/Silver/2606. 바이러스/바이러스.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');

const N = +input[0];
const adj = Array.from({length: N+1}, () => []);

input.slice(2).map(line => line.split(' ').map(Number)).forEach(([from, to]) => {
  adj[from].push(to);
  adj[to].push(from);
})

let ret = 0;
const dfs = (here, visited = new Set()) => {
  visited.add(here);

  for(let there of adj[here]){
    if(visited.has(there)) continue;
    ret++;
    dfs(there, visited);
  }
  return;
}

const answer = () => {
  dfs(1);
  return ret;
}

console.log(answer());